third_party/spdk_defs/nvme_defs.h

```diff
@@ -192,7 +192,36 @@ enum class NvmOpcode : uint8_t {
   kSctVendorSpecific = 0x7,
 };
 
-} // namespace nvme_defs
+// NVMe Base Specification Figure 15
+// https://nvmexpress.org/wp-content/uploads/NVM-Express-1_0e.pdf
+struct Status {
+  uint16_t p : 1;    // phase tag
+  uint16_t sc : 8;   // status code
+  uint16_t sct : 3;  // status code type
+  uint16_t rsvd2 : 2;
+  uint16_t m : 1;    // more
+  uint16_t dnr : 1;  // do not retry
+};
+static_assert(sizeof(Status) == 2);
+
+// Completion Queue Entry
+// NVMe Base Specification Figure 12
+// https://nvmexpress.org/wp-content/uploads/NVM-Express-1_0e.pdf
+struct GenericQueueEntryCpl {
+  // dword 0
+  uint32_t cdw0;  // command-specific
+  // dword 1
+  uint32_t rsvd1;
+  // dword 2
+  uint16_t sqhd;  // submission queue head pointer
+  uint16_t sqid;  // submission queue identifier
+  // dword 3
+  uint16_t cid;  // command identifier
+  spdk_nvme_status status;
+};
```

Comment:
Are we not doing struct packing for NVMe?

Reply:
This particular structure doesn't need to be packed because it has no bit fields & has a size which is a multiple of it's largest type. I can add explicit bit fields and a packing tag if we feel it's more consistent, but when the structure has nothing fancy going on to begin with I think we might consider making an exception

Reply:
imo we should do it for consistency. Although others might feel differently because it doesn't make a difference on the compiler level.

Reply:
I'm not sure a well-defined memory layout is guaranteed unless packing is explicitly requested, so I'd favor doing at least that much.